File: post.go
package user

type Post struct {
	Id              int    `json:"id" db:"id"`
	UserId          int    `json:"userId" db:"user_id" binding:"required"`
	Text            string `json:"text" db:"text" binding:"required"`
	IsModerated     bool   `json:"isModerated" db:"is_moderated" binding:"required"`
	PublicationTime string `json:"publicationTime" db:"publication_time"`
	Tags            []Tag  `json:"tags" db:"tags_id"`
}

type OrderType int8

const (
	Ascending OrderType = iota
	Descending
)

type PostFilter struct {
	TagsList []int     `json:"tagsList"`
	OrderBy  OrderType `json:"orderBy"`
}

type Tag struct {
	Id    int    `json:"id" db:"id"`
	Title string `json:"title" db:"title" binding:"required"`
}

File: client/src/app/shared/models/Post.ts
import {IInterest} from "./Interest";

export interface IPost {
  author: any;
  userId: number
  text: string
  publicationTime?: Date
  tags: IInterest[]
}

export class Post {
  id: number;
  author: any;
  userId: number
  text: string
  publicationTime?: Date
  tags: IInterest[]
}
